### PR TITLE
Add Trino 407 release notes

### LINKED ISSUE #15854: Release notes for 407

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 407 release. Fixes #15854 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

## Verification for each pull request

Format: maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 25 Jan 2023

@ebyhr #15806 ✅ rn ✅ docs
@ebyhr #15852 ✅ rn ✅ docs
@ebyhr #15851 ✅ rn ✅ docs

## 26 Jan 2023

@findepi #15350 ✅ rn ✅ docs
@raunaqmorarka #15713 ✅ rn ✅ docs
@ebyhr #15807 ✅ rn ✅ docs
@ebyhr #15857 ✅ rn ✅ docs
@findepi #15841 ✅ rn ✅ docs
@findepi #15844 ✅ rn ✅ docs
@sopel39 #15838 ✅ rn ✅ docs
@electrum #15703 ✅ rn ✅ docs
@findepi #15843 ✅ rn ✅ docs
@findepi #15825 ✅ rn ✅ docs
@findepi #15863 ✅ rn ✅ docs
@ebyhr #15855 ✅ rn ✅ docs
@ebyhr #15866 ✅ rn ✅ docs

## 27 Jan 2023

@findepi #15849 ✅ rn ✅ docs
@findepi #15862 ✅ rn ✅ docs
@findepi #15861 ✅ rn ✅ docs
@findepi #15744 ✅ rn ✅ docs
@findepi #15860 ✅ rn ✅ docs
@sopel39 #15875 ✅ rn ✅ docs
@martint #15085 ✅ rn ✅ docs
@ebyhr #15876 ✅ rn ✅ docs

## 28 Jan 2023

@martint #15886 ✅ rn ✅ docs
@electrum #15833 ✅ rn ✅ docs

## 30 Jan 2023

@findepi #15879 ✅ rn ✅ docs
@raunaqmorarka #15850 ✅ rn ✅ docs
@kasiafi #14566 ✅ rn ✅ docs
@kokosing #15871 ✅ rn ✅ docs
@sopel39 #15864 ✅ rn ✅ docs
@arhimondr #15721 ✅ rn ❌ docs
@arhimondr #15901 ✅ rn ✅ docs
@ebyhr #15889 ✅ rn ✅ docs
@ebyhr #15902 ✅ rn ✅ docs
@raunaqmorarka #15858 ✅ rn ✅ docs

## 31 Jan 2023

@Praveen2112 #15823 ✅ rn ✅ docs
@ebyhr #15906 ✅ rn ✅ docs
@ebyhr #15905 ✅ rn ✅ docs
@findepi #15896 ✅ rn ✅ docs
@sopel39 #15811 ✅ rn ✅ docs
@sopel39 #15899 ✅ rn ✅ docs
@ebyhr #15436 ✅ rn ✅ docs
@ebyhr #15094 ✅ rn ❌ docs
@raunaqmorarka #15897 ✅ rn ✅ docs
@ebyhr #15914 ✅ rn ✅ docs

## 1 Feb 2023

@findepi #15909 ✅ rn ✅ docs
@findepi #15912 ✅ rn ✅ docs
@findepi #15908 ✅ rn ✅ docs
@sopel39 #15579 ✅ rn ✅ docs
@sopel39 #15924 ✅ rn ✅ docs
@dain #15775 ✅ rn ✅ docs
@losipiuk #15870 ✅ rn ✅ docs
@ebyhr #15922 ✅ rn ✅ docs

## 2 Feb 2023

@findepi #15929 ✅ rn ✅ docs
@raunaqmorarka #15923 ✅ rn ✅ docs
@findepi #15939 ✅ rn ✅ docs
@findepi #15936 ✅ rn ✅ docs
@raunaqmorarka #15942 ✅ rn ✅ docs
@ebyhr #15958 ✅ rn ✅ docs

## 3 Feb 2023

@sopel39 #15489 ✅ rn ❌ docs
@sopel39 #15948 ✅ rn ✅ docs
@ebyhr #15856 ✅ rn ✅ docs
@findepi #15957 ✅ rn ✅ docs
@ebyhr #15959 ✅ rn ✅ docs
@ebyhr #15940 ✅ rn ✅ docs
@sopel39 #15949 ✅ rn ✅ docs
@sopel39 #15839 ✅ rn ✅ docs

## 4 Feb 2023

@ebyhr #15893 ✅ rn ✅ docs

## 5 Feb 2023

@kokosing #15781 ✅ rn ✅ docs
@ebyhr #15747 ✅ rn ✅ docs

## 6 Feb 2023

@raunaqmorarka #15954 ✅ rn ✅ docs
@ebyhr #15396 ✅ rn ✅ docs
@sopel39 #15983 ✅ rn ✅ docs
@findepi #15985 ✅ rn ✅ docs
@electrum #15702 ✅ rn ✅ docs
@electrum #14020 ✅ rn ✅ docs
@electrum #15765 ✅ rn ✅ docs
@findepi #15638 ✅ rn ✅ docs
@findepi #15969 ✅ rn ✅ docs
@raunaqmorarka #13814 ✅ rn ✅ docs

## 7 Feb 2023

@losipiuk #15947 ✅ rn ✅ docs
@findepi #15878 ✅ rn ✅ docs
@raunaqmorarka #16001 ✅ rn ✅ docs
@raunaqmorarka #15816 ✅ rn ✅ docs
@kasiafi #15989 ✅ rn ✅ docs
@findepi #15981 ✅ rn ✅ docs
@ebyhr #15996 ✅ rn ✅ docs
@Praveen2112 #15935 ✅ rn ✅ docs

## 8 Feb 2023

@findepi #15984 ✅ rn ✅ docs
@findepi #15453 ✅ rn ✅ docs
@findepi #14373 ✅ rn ✅ docs
@martint #16000 ✅ rn ✅ docs
@kasiafi #15994 ✅ rn ✅ docs
@arhimondr #15933 ✅ rn ✅ docs
@findepi #16017 ✅ rn ✅ docs

## 9 Feb 2023

@kasiafi #16010 ✅ rn ✅ docs
@hashhar #15724 ✅ rn ✅ docs
@findepi #15797 ✅ rn ✅ docs
@kokosing #15941 ✅ rn ✅ docs
@ebyhr #16044 ✅ rn ✅ docs
@electrum #16049 ✅ rn ✅ docs
@kasiafi #16014 ✅ rn ✅ docs
@electrum #15577 ✅ rn ✅ docs
@findepi #16043 ✅ rn ✅ docs
@electrum #14402 ✅ rn ✅ docs
@arhimondr #16031 ✅ rn ✅ docs
@ebyhr #16028 ✅ rn ✅ docs
@raunaqmorarka #15883 ✅ rn ✅ docs
@raunaqmorarka #15962 ✅ rn ✅ docs
@Praveen2112 #15960 ✅ rn ✅ docs

## 10 Feb 2023

@ebyhr #16040 ✅ rn ✅ docs
@ebyhr #16069 ✅ rn ✅ docs

## 11 Feb 2023

@ebyhr #16032 ✅ rn ✅ docs
@kasiafi #16058 ✅ rn ✅ docs

## 12 Feb 2023

@ebyhr #16047 ✅ rn ✅ docs
@raunaqmorarka #15986 ✅ rn ✅ docs
@ebyhr #16062 ✅ rn ✅ docs
@ebyhr #15788 ✅ rn ✅ docs

## 13 Feb 2023

@arhimondr #15917 ✅ rn ✅ docs
@ebyhr #16097 ✅ rn ✅ docs

## 14 Feb 2023

@kokosing #16060 ✅ rn ✅ docs
@Praveen2112 #15898 ✅ rn ✅ docs
@ebyhr #15956 ✅ rn ✅ docs
@arhimondr #16111 ✅ rn ✅ docs
@ebyhr #15874 ✅ rn ✅ docs
@ebyhr #16108 ✅ rn ✅ docs
@ebyhr #16117 ✅ rn ✅ docs
@ebyhr #15556 ✅ rn ✅ docs
@ebyhr #16118 ✅ rn ✅ docs
@raunaqmorarka #16052 ✅ rn ✅ docs

## 15 Feb 2023

@ebyhr #16007 ✅ rn ✅ docs
@ebyhr #15973 ✅ rn ✅ docs
@raunaqmorarka #16110 ✅ rn ✅ docs
@ebyhr #16055 ✅ rn ✅ docs
@kokosing #16121 ✅ rn ✅ docs
@kokosing #16086 ✅ rn ✅ docs
@arhimondr #16113 ✅ rn ✅ docs
@electrum #16133 ✅ rn ❌ docs
@ebyhr #14869 ✅ rn ✅ docs

## 16 Feb 2023

@martint #16134 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-407.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Release 407 (1 Feb 2023)
+
+## General
+
+* Add support for adaptive task request sizing, which can be enabled with the
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:5]:
```suggestion
* Add support for adaptive task request sizing. It can be enabled with the
```

File: docs/src/main/sphinx/release/release-407.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  `query.remote-task.enable-adaptive-request-size` configuration property and
+  configured with the `query.remote-task.max-request-size`,
+  `query.remote-task.request-size-headroom`, and
+  `query.remote-task.guaranteed-splits-per-task` configuration properties or
+  their respective session properties. ({issue}`15721`)
+* Improve query performance when a predicate always evaluates to a null value. ({issue}`15744`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:11]: Is "always" needed here?

File: docs/src/main/sphinx/release/release-407.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+* Improve performance of queries with joins on the output of global
+  aggregations. ({issue}`15858`)
+* Fix precision loss when converting `time` values with a precision higher than
+  3 and `time with time zone` values with lower precision. ({issue}`15861`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:15]: maybe use "three"

**Review by mosabua — APPROVED** [review]

**Comment by colebow** [comment]
Looks like we're blocked by #15931

**Comment by arhimondr** [comment]
Here is the fix: https://github.com/trinodb/trino/pull/15933

File: docs/src/main/sphinx/release/release-407.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+* Remove support for the legacy, non-optimized Parquet writer. ({issue}`15436`)
+* Improve performance of reading decimal columns from Parquet files. ({issue}`15713`)
+* Improve performance of reading numeric types from Parquet files. ({issue}`15850`)
+
+## Hive connector
+
+* Improve performance of reading decimal columns from Parquet files. ({issue}`15713`)
+* Improve performance of reading numeric types from Parquet files. ({issue}`15850`)
+* Reduce query latency. ({issue}`15811`)
```
Comment (ebyhr) [inline docs/src/main/sphinx/release/release-407.md:29]: I think #15811 reduces query latency in Delta Lake connector as well. 
cc: @sopel39 @findepi

Comment (findepi) [reply docs/src/main/sphinx/release/release-407.md:29]: I don't think it does. Delta connector doesn't read stats from the metastore.

File: docs/src/main/sphinx/release/release-407.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Release 407 (1 Feb 2023)
+# Release 407 (15 Feb 2023)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:1]: 13

File: docs/src/main/sphinx/release/release-407.md [hunk 6]
```diff
@@ -8,37 +8,96 @@
...
 * Improve query performance when a predicate evaluates to a null value. ({issue}`15744`)
 * Improve performance of queries with joins on the output of global
   aggregations. ({issue}`15858`)
-* Fix precision loss when converting `time` values with a precision higher than
-  three and `time with time zone` values with lower precision. ({issue}`15861`)
+* Improve performance of selective queries, queries that read a small number of
+  columns, and queries that process tables with large Parquet row groups or Orc
+  stripes. ({issue}`15579`)
+* Improve concurrency when small, concurrent queries are run on a large cluster.
+  Trino now automatically determines many nodes to use for distributed joins
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:20]: how many nodes ??

File: docs/src/main/sphinx/release/release-407.md [hunk 7]
```diff
@@ -8,37 +8,96 @@
...
+  and aggregations. This is managed by the `query.max-hash-partition-count`
+  (renamed from `query.hash-partition-count`) and `query.min-hash-partition-count`
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:22]: Is the docs updated?

File: docs/src/main/sphinx/release/release-407.md [hunk 8]
```diff
@@ -8,37 +8,96 @@
...
+  configuration properties. You can also now set the minimum number of rows and
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:23]: I am not a friend of using "now" .. especially not twice..

Comment (mosabua) [reply docs/src/main/sphinx/release/release-407.md:23]: Could this even be a separate release notes entry .. but with the same PR number?

File: docs/src/main/sphinx/release/release-407.md [hunk 9]
```diff
@@ -8,37 +8,96 @@
...
+  minimum data size per task using the `optimizer.min-input-rows-per-task` and
+  `optimizer.min-input-size-per-task` configuration properties. ({issue}`15489`)
+* Improve performance of queries with window functions. ({issue}`15994`)
 * Change the `table_type` of materialized views in `information_schema.tables`
   from `BASE TABLE` to `MATERIALIZED VIEW`. ({issue}`15350`)
+* Fix precision loss when converting `time` values with a precision higher than
+  three and `time with time zone` values with lower precision. ({issue}`15861`)
+* Fix handling of `UNKNOWN` type for row type in exchange operator. ({issue}`15948`)
+* Fix potential incorrect results when a query fails but reports normal
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:32]: fails, but

File: docs/src/main/sphinx/release/release-407.md [hunk 10]
```diff
@@ -8,37 +8,96 @@
...
+* Improve performance of reading timestamp and boolean types from Parquet files. ({issue}`15954`)
+* Prevent creating a table when the specified schema does not exist. ({issue}`15779`)
+* Fix query failure when reading Parquet files written by Impala. ({issue}`15942`)
+* Fix listing relations failure when a Glue table has no table type set. ({issue}`15909`)
+* Fix failure when encountering access denied exceptions while listing
+  materialized views in the Glue metastore. ({issue}`15893`)
+
+## Kudu connector
+
+* Fix authentication failure when Kerberos tickets expire. ({issue}`14372`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-407.md:90]: is it really called a Kerberos ticket ?

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-407.md (hunk 7 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-407.md:22]: No. ☹️

File: docs/src/main/sphinx/release/release-407.md (hunk 9 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-407.md:32]: "reports normal completion" isn't an independent clause, so this comma would be incorrect

File: docs/src/main/sphinx/release/release-407.md (hunk 10 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-407.md:90]: [Yep!](https://web.mit.edu/kerberos/krb5-latest/doc/user/tkt_mgmt.html)

Comment (mosabua) [reply docs/src/main/sphinx/release/release-407.md:91]: Wow.. crazy

File: docs/src/main/sphinx/release/release-407.md (hunk 7 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-407.md]: Sign. .. can you send a PR ?

File: docs/src/main/sphinx/release/release-407.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+## General
+
+* Add support for adaptive task request sizing. It can be enabled with the
+  `query.remote-task.enable-adaptive-request-size` configuration property and
+  configured with the `query.remote-task.max-request-size`,
+  `query.remote-task.request-size-headroom`, and
+  `query.remote-task.guaranteed-splits-per-task` configuration properties or
+  their respective session properties. ({issue}`15721`)
+* Add support for correlated queries with single-row values. ({issue}`15989`)
+* Add support for configuring the minimum number of rows and
```
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:12]: Has the default changed? What's the user visible impact? If this is not visible, remove the entry. 

Also, why are these "optimizer.xxx" options? The seem to be related to the runtime, in which case they should be "query.xxxx". cc @gaurav8297 @sopel39

File: docs/src/main/sphinx/release/release-407.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+  configuration properties. ({issue}`15489`)
+* Improve query memory tracking. ({issue}`15983`)
+* Improve query performance when a predicate evaluates to a null value. ({issue}`15744`)
+* Improve performance of queries with joins on the output of global
+  aggregations. ({issue}`15858`)
+* Improve performance of selective queries, queries that read a small number of
+  columns, and queries that process tables with large Parquet row groups or Orc
+  stripes. ({issue}`15579`)
+* Improve performance of queries with window functions. ({issue}`15994`)
+* Change the `table_type` of materialized views in `information_schema.tables`
```
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:28]: @findepi , why did this change? We discussed this being incompatible with the SQL spec, so it shouldn't have gone in.

File: docs/src/main/sphinx/release/release-407.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+  from `BASE TABLE` to `MATERIALIZED VIEW`. ({issue}`15350`)
+* Fix precision loss when converting `time` values with a precision higher than
+  three and `time with time zone` values with lower precision. ({issue}`15861`)
+* Fix handling of `UNKNOWN` type for row type in exchange operator. ({issue}`15948`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:30]: UNKNOWN is not a type. If anything, this should say "when processing null values". 

However, the entry should be talking about the user-visible impact. What does "Fix handling" mean? Was there a failure? A performance problem? When does it manifest? What kinds of queries trigger it?

cc @lukasz-stec @sopel39

File: docs/src/main/sphinx/release/release-407.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+
+* Add support for the `unregister_table` procedure. ({issue}`15784`)
+* Add support for inserting into tables that have `CHECK` constraints. ({issue}`15396`)
+* Add support for writing to the [change data feed](https://docs.delta.io/2.0.0/delta-change-data-feed.html). ({issue}`15453`)
+* Improve performance of reading decimal types from Parquet files. ({issue}`15713`)
+* Improve performance of reading numeric types from Parquet files. ({issue}`15850`)
+* Improve performance of reading string types from Parquet files. ({issue}`15897`, {issue}`15923`)
+* Improve performance of reading timestamp and boolean types from Parquet files. ({issue}`15954`)
+* Improve query performance on tables created by Trino with `CREATE TABLE AS`. ({issue}`15878`)
+* Remove support for the legacy, non-optimized Parquet writer. ({issue}`15436`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:44]: Remove "non-optimized". Just say "the legacy Parquet writer".

File: docs/src/main/sphinx/release/release-407.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+* Fix listing relations failure when a Glue table has no table type set. ({issue}`15909`)
+
+## Hive connector
+
+* Reduce query latency. ({issue}`15811`)
+* Improve performance of reading decimal types from Parquet files. ({issue}`15713`)
+* Improve performance of reading numeric types from Parquet files. ({issue}`15850`)
+* Improve performance of reading string types from Parquet files. ({issue}`15897`, {issue}`15923`)
+* Improve performance of reading timestamp and boolean types from Parquet files. ({issue}`15954`)
+* Fix query failure when reading Parquet files written by Impala. ({issue}`15942`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:55]: The proper name is "Apache Impala".

File: docs/src/main/sphinx/release/release-407.md [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+
+## Iceberg connector
+
+* Add support for the `unregister_table` procedure. ({issue}`15784`)
+* Add the `iceberg.jdbc-catalog.connection-user` and
```
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:69]: What does "adding xxx property to the JDBC catalog"? What's the functionality being added? Should this be "Add support for specifying a user and password when connecting to the JDBC catalog"?

cc @ebyhr

File: docs/src/main/sphinx/release/release-407.md (hunk 13 above)
Comment (lukasz-stec) [reply docs/src/main/sphinx/release/release-407.md:30]: AFAIK, there are no real queries that could trigger this in the current code. I would drop the release note.

File: docs/src/main/sphinx/release/release-407.md (hunk 12 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-407.md:26]: Can we revert to unblock the release, then discuss this later?

Comment (mosabua) [reply docs/src/main/sphinx/release/release-407.md:26]: I agree with reverting for now so we can get a release out.

Comment (mosabua) [reply docs/src/main/sphinx/release/release-407.md:26]: cc @jhlodin

**Review by mosabua — APPROVED** [review]
All looks good. A bunch of things are probably missing documentation updates though.

File: docs/src/main/sphinx/release/release-407.md [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+  by automatically determining how many nodes to use for distributed joins
+  and aggregations. This can be managed with the `query.max-hash-partition-count`
+  (renamed from `query.hash-partition-count`) and `query.min-hash-partition-count`
+  configuration properties. ({issue}`15489`)
+* Improve query memory tracking. ({issue}`15983`)
+* Improve memory usage accounting for queries with dynamic filters. ({issue}`16110`)
+* Improve query performance when a predicate evaluates to a null value. ({issue}`15744`)
+* Improve performance of queries with joins on the output of global
+  aggregations. ({issue}`15858`)
+* Improve performance of selective queries, queries that read a small number of
```
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:23]: ORC

File: docs/src/main/sphinx/release/release-407.md [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+  columns, and queries that process tables with large Parquet row groups or Orc
+  stripes. ({issue}`15579`)
+* Improve performance of queries with window functions. ({issue}`15994`)
+* Return an exit code of `100` when Trino crashes during startup. ({issue}`16113`)
+* Fix precision loss when converting `time` values with a precision higher than
+  three and `time with time zone` values with lower precision. ({issue}`15861`)
+* Fix potential incorrect results when a query fails but reports normal
```
Comment (martint) [inline docs/src/main/sphinx/release/release-407.md:30]: Why would a query fail and report normal completion? Or is the bug that it fails and reports normal completion?

If so, I'd rephrase as: "Fix potential incorrect results due to query reporting normal completion instead of failing"
